python/enthic/compute_stats.py

```diff
@@ -0,0 +1,234 @@
+                    break
+
+        # If official value match computed value by adding all children's value (no substraction) with less than 0.5% error
+        elif abs((computed_sum_without_sign - value) / value) < relative_error or abs(computed_sum_without_sign - value) < absolute_error:
+            tree_item['data']['status'] = "checked"
+            # Fix children sign and/or set to zero missing values if any
+            for child_name in tree_item['children']:
+                flip_sign(tree_item['children'][child_name])
+                set_to_zero_computed(tree_item['children'][child_name])
+        elif abs((computed_sum - value) / value) > 100:
+            tree_item['data']['status'] = "error"
+        else:
+            tree_item['data']['status'] = "error"
+
+    if computed_sum != tree_item['data']['value']:
+        tree_item['data']['computedValue'] = computed_sum_from_computed
+
+def set_to_zero_computed(tree_item):
+    """
+    set value of given item and it's children to zero when it hasn't another value
+        :param tree_item : item to set to zero
+    """
+    if math.isnan(tree_item['data']['value']) and ('computedValue' not in tree_item['data'] or tree_item['data']['computedValue'] == 0):
+        tree_item['data']['value'] = 0
+        tree_item['data']['status'] = "computed"
+        if 'children' in tree_item:
+            for child_name in tree_item['children']:
+                set_to_zero_computed(tree_item['children'][child_name])
+
+def flip_sign(item):
+    """
+    Flip sign of given item
+        :param item : item to flip
+    """
+    if item['sign'] == -1:
+        item['data']['value'] = -item['data']['value']
+        item['data']['status'] = "signFlipped"
+
+def compute_annual_share_score(tree):
+    """
+    Computes the share score of given data
+        :param tree: data from which compute the score
+    """
+    # Retrieve needed values
+    participation = tree['children']['ParticipationSalariesAuxResultats']['data']['value']
+    impot = tree['children']['ImpotsSurLesBenefices']['data']['value']
+    resultat_financier = tree['children']['ResultatExceptionnel']['data']['value']
+    resultat_exceptionnel = tree['children']['ResultatAvantImpot']['children']['ResultatFinancier']['data']['value']
+    resultat_exploitation = tree['children']['ResultatAvantImpot']['children']['ResultatExploitation']['data']['value']
+
+    # If any values unknown, cannot compute score
+    if any(math.isnan(value) for value in [participation, impot, resultat_financier, resultat_exceptionnel, resultat_exploitation]):
+        return float('nan')
+
+    # Compute score if possible
+    shared_part = participation + impot
+    resultat_capitaux = resultat_financier + resultat_exceptionnel
+    if shared_part >= 0:
+        if resultat_capitaux <= 0 and resultat_exploitation != 0:
+            return shared_part / resultat_exploitation
+
+    return float('nan')
+
```

Comment:
PEP 8: E302 expected 2 blank lines, found 1, use autopep 8 or an IDE that reformat according to PEP8 e.g. Pycharm Ctrl+Alt+L .